woocommerce/woocommerce
# Add RC draft post command, make some minor improvements to beta post command

### All Submissions:






### Changes proposed in this Pull Request:

As a follow up to #37142 this adds a very similar command to make the RC post. Like the beta post I've kept it very simple for now to avoid complications.

The goal is to avoid a lot of the copy/pasting involved in creating a new post and get you very close to a complete post. Later we can enhance this to add the change list.

I also back ported some minor improvements to output and documentation to the beta post generator (such as outputting an edit URL to save you some more time.) And fixing a bug where we didn't validate that the version passed has the correct type (e.g. running `rc` command you cannot pass a beta version)



### How to test the changes in this Pull Request:




1. First you'll need to check out this branch and ensure that `pnpm i` has been run on the monorepo
2. You'll also need an oauth app set up on developer.wordpress.com. See [the Readme for instructions](https://github.com/woocommerce/woocommerce/blob/e43bdcca60c44c87f9ef0fa9000a7424355b368a/tools/release-posts/README.md?plain=1#L17) and make sure as part of that you set the secrets required in a local `.env` file.
2. Traverse to the release post tool: `cd tools/release-posts`
3. Now run a test command for a fictitious RC post (don't worry even if you don't pass `--outputOnly` it will only post a draft
4. example command: `pnpm release-post rc 7.5.0-rc.1 --outputOnly`. See if you like the output
5. now to get a better idea of how the output looks, post a draft: `pnpm release-post rc 7.5.0-rc.1`
6. you can also test that it rejects a beta to rc command and rejects a rc to beta command. e.g. this should produce an error: `pnpm release-post rc 7.5.0-beta.1`

Note that during the command it will show you a list of posts it searched to link the last release post. I don't expect this to be perfect and it will sometimes not find something suitable but the idea is best effort and you can edit it manually later if it didn't find a good post.



### Other information:




### FOR PR REVIEWER ONLY:
